src/v/raft/consensus.cc

```diff
@@ -795,8 +795,11 @@ ss::future<> consensus::start() {
                */
               if (st.config_batches_seen() > 0) {
                   f = f.then([this, st = std::move(st)]() mutable {
+                      auto cfg = st.release_config();
+                      // NOTE: may be removed in future versions
+                      cfg.maybe_set_initial_revision(_self.revision());
                       return _configuration_manager.add(
```

Comment:
can we audit every line of `_configuration_manager*` should be only a handful and make sure that versions are all up to date too.

Reply:
i think we should make the `cfg.maybe_set_initial_version()` as part of `_configuration_manager.add()` method. 
so it works in all places.